service/src/main/java/uk/gov/hmcts/reform/fpl/service/JudicialService.java

```diff
@@ -449,53 +436,93 @@ public void cleanupHearingRoles(Long caseId) {
-                    + " details manually.");
+        caseDetails.getData().put("judgeAndLegalAdvisor", hearingJudge);
+        return List.of();
+    }
+
+    private ZonedDateTime currentTimeUK() {
+        return ZonedDateTime.now(LONDON_TIMEZONE);
+    }
+
+    public Judge buildAllocatedJudgeFromEventData(AllocateJudgeEventData eventData) {
+        if (isJudgeDetailsFromJUP(eventData)) {
+            if (isEmpty(eventData.getJudicialUser()) || isEmpty(eventData.getJudicialUser().getPersonalCode())) {
+                return null;
             }
+
+            return this.getJudge(eventData.getJudicialUser().getPersonalCode())
+                .map(judicialUserProfile -> Judge.fromJudicialUserProfile(judicialUserProfile,
+                        (FEE_PAID_JUDGE.equals(eventData.getJudgeType())) ? eventData.getFeePaidJudgeTitle() : null)
+                    .toBuilder()
+                    .judgeType(eventData.getJudgeType())
+                    .build()
+                )
+                .orElse(null);
         } else {
-            // entered the judge manually - lookup in our mappings and add UUID manually
-            Optional<String> possibleId = this
-                .getJudgeUserIdFromEmail(caseData.getHearingJudge().getJudgeEmailAddress());
+            Judge manualJudgeDetails = eventData.getManualJudgeDetails()
+                .toBuilder()
+                .judgeType(eventData.getJudgeType())
+                .build();
+
+            // entering manually, check against our lookup tables, they may be a legal adviser
+            Optional<String> possibleId = this.getJudgeUserIdFromEmail(manualJudgeDetails.getJudgeEmailAddress());
 
+            // if they are in our maps - add their UUID extra info to the case
             if (possibleId.isPresent()) {
-                // we can manually assign the role again based on our knowledge of JRD/SRD
-                hearingJudge = JudgeAndLegalAdvisor.from(caseData.getHearingJudge()).toBuilder()
+                return manualJudgeDetails.toBuilder()
                     .judgeJudicialUser(JudicialUser.builder()
                         .idamId(possibleId.get())
                         .build())
-                    .legalAdvisorName(caseData.getLegalAdvisorName())
                     .build();
             } else {
-                // We cannot assign manually, just have to leave the judge as is.
-                hearingJudge = JudgeAndLegalAdvisor.from(caseData.getHearingJudge()).toBuilder()
-                    .legalAdvisorName(caseData.getLegalAdvisorName())
-                    .build();
+                if (MAGISTRATES.equals(manualJudgeDetails.getJudgeTitle())) {
+                    return manualJudgeDetails.toBuilder().judgeLastName(null).build();
+                } else {
+                    return manualJudgeDetails.toBuilder().judgeFullName(null).build();
+                }
```

Comment:
looks like these return statements are equivalent - so can probably remove the conditional?

Reply:
Here is cleaning up the last name or full name fields